aspnetcore/blazor/lifecycle.md

```diff
@@ -0,0 +1,175 @@
+
+For a synchronous operation, use `OnInitialized`:
+
+```csharp
+protected override void OnInitialized()
+{
+    ...
+}
+```
+
+### Before parameters are set
+
+<xref:Microsoft.AspNetCore.Components.ComponentBase.SetParametersAsync*> sets parameters supplied by the component's parent in the render tree:
+
+```csharp
+public override async Task SetParametersAsync(ParameterView parameters)
+{
+    await ...
+
+    await base.SetParametersAsync(parameters);
+}
+```
+
+<xref:Microsoft.AspNetCore.Components.ParameterView> contains the entire set of parameter values each time `SetParametersAsync` is called.
+
+The default implementation of `SetParametersAsync` sets the value of each property decorated with the `[Parameter]` or `[CascadingParameter]` attribute that has a corresponding value in the `ParameterView`. Parameters that don't have a corresponding value in `ParameterView` are left unchanged.
+
+If `base.SetParametersAync` isn't invoked, the custom code can interpret the incoming parameters value in any way required. For example, there's no requirement to assign the incoming parameters to the properties on the class.
+
+### After parameters are set
+
+<xref:Microsoft.AspNetCore.Components.ComponentBase.OnParametersSetAsync*> and <xref:Microsoft.AspNetCore.Components.ComponentBase.OnParametersSet*> are called when a component has received parameters from its parent and the values are assigned to properties. These methods are executed after component initialization and each time new parameter values are specified:
+
+```csharp
+protected override async Task OnParametersSetAsync()
+{
+    await ...
+}
+```
+
+> [!NOTE]
+> Asynchronous work when applying parameters and property values must occur during the `OnParametersSetAsync` lifecycle event.
+
+```csharp
+protected override void OnParametersSet()
+{
+    ...
+}
+```
+
+### After component render
+
+<xref:Microsoft.AspNetCore.Components.ComponentBase.OnAfterRenderAsync*> and <xref:Microsoft.AspNetCore.Components.ComponentBase.OnAfterRender*> are called after a component has finished rendering. Element and component references are populated at this point. Use this stage to perform additional initialization steps using the rendered content, such as activating third-party JavaScript libraries that operate on the rendered DOM elements.
+
+The `firstRender` parameter for `OnAfterRenderAsync` and `OnAfterRender`:
+
+* Is set to `true` the first time that the component instance is invoked.
+* Ensures that initialization work is only performed once.
+
+```csharp
+protected override async Task OnAfterRenderAsync(bool firstRender)
+{
+    if (firstRender)
+    {
+        await ...
+    }
+}
+```
+
+> [!NOTE]
+> Asynchronous work immediately after rendering must occur during the `OnAfterRenderAsync` lifecycle event.
+>
+> A component doesn't rerender after the completion of any returned <xref:System.Threading.Tasks.Task>. Infinite render loops are avoided.
```

Comment:
I know you have a rule that things have to be phrased like this, but it's nearly impossible to understand. It's so ambiguous.

```suggestion
> Even if you return a <xref:System.Threading.Tasks.Task> from `OnAfterRenderAsync`, the framework will not schedule a further render cycle for your component once that task completes. This is to avoid an infinite render loop. It's different from the other lifecycle methods, which do schedule a further render cycle once the returned task completes.
```

Reply:
It's cool ... I'll just change the future tense.